README.md

```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
+There are two ways to configure a Hazelcast Node.js Client:
+- Programmatically
+- Declaratively (JSON)
+
+This section describes some network configuration settings to cover common use cases in connecting the client to a cluster. Refer to (CONFIG.md)
+and the following sections for information about detailed network configuration and/or additional features of Hazelcast Node.js Client
+configuration.
+
+An easy way to configure your Hazelcast Node.js Client is to create a `Config` object and set the appropriate options. Then you can
+supply this object to your client at the startup. Another way to configure your client is to provide a `hazelcast-client.json` file. This approach is similar to `hazelcast.xml` approach
+in configuring the member. Note that `hazelcast-client.json` is a JSON file whereas member configuration is XML based. Although these
+two formats are different, you will realize that the names of the configuration parameters are the same for both the client and member.
+It is done this way to make it easier to transfer Hazelcast skills to multiple platforms.
+
+Once you embedded hazelcast-client to your Node.js project, you may follow any of programmatic or declarative configuration approaches.
+We will provide both ways for each configuration option in this section. Pick one way and stick to it.
+
+---
+
+**Programmatic configuration**
+You need to create a `ClientConfig` object and adjust its properties. Then you can pass this object to the client when starting it.
+
+```javascript
+let Client = require('hazelcast-client').Client;
+let Config = require('hazelcast-client').Config;
+let cfg = new Config.ClientConfig();
+Client.newHazelcastClient(cfg)
+```
+
+**Declarative configuration**
+Hazelcast Node.js Client looks for a `hazelcast-client.json` in the current working directory unless you provide a configuration object
+at the startup. If you intend to configure your client using a configuration file, then place a `hazelcast-client.json` in the directory
+of your application's entry point.
+
+If you prefer to keep your `hazelcast-client.json` file somewhere else, you can override the environment variable `HAZELCAST_CLIENT_CONFIG`
+with the location of your config file. In this case, the client uses the configuration file specified in the environment variable.
+
+For the structure of `hazelcast-client.json`, take a look at [hazelcast-client-full.json](test/config/hazelcast-client-full.json). You
+can use only the relevant parts of the file in your `hazelcast-client.json` and remove the rest. Default configuration is used for any
+part that you do not explicitly set in `hazelcast-client.json`.
+
+---
+
+If you run Hazelcast IMDG members in a different server than the client, you most probably have configured the members' ports and cluster
+names as explained in the previous section. If you did, then you need to make certain changes to the network settings of your client.
+
+### Group Settings
+
+- Programmatic
```

Comment:
Using this style is better to be compatible with other PRs.
**Programmatic:**
```javascript
let cfg = new Config.ClientConfig();
cfg.group.name = // group name of you cluster
```

**Declarative:**
```json
{
    "group": {
        "name": "group name of you cluster"
    }
}
```